parser/src/lexer.rs

```diff
@@ -914,7 +918,9 @@ where
                 }
                 Some(']') => {
                     let result = self.eat_single_char(Tok::Rsqb);
-                    self.nesting -= 1;
+                    if self.closing_bracket_is_valid() {
```

Comment:
I wouldn't use this function. For me it's less clear and longer than the code it replaces.

Reply:
Ok, what should I do? Should I create a new pull request from scratch or simply make a new commit?

Reply:
I'd continue to push to this branch. Once it's ready to go we can rebase to clean up. (I presume you can force push here, not sure how github permissions work.)

Reply:
Ok I pushed the commit, what should I do now to clean the history without  messing everything up?

Reply:
It looks like under the circumstances I can use github squash-merge to achieve something similar, so I'll just do that and you don't need to do anything.

Thanks for your work on this. Looking good now.

My approach would be:

```
git rebase -i HEAD~6
```

(I'd normally use ```git rebase -i $(git merge-base HEAD origin/master)```, but I think origin/master will be the same as master in your case, so I'm just making it easy and saying the last six commits.)

Change first commit to reword, all subsequent ones to fixup, then:

```
git push -f
```

Reply:
And I typo'd the commit message. :(